shell/utils/router.js

```diff
@@ -89,14 +89,13 @@ export const routeMatched = (to, fn) => {
 };
 
 /**
- * Given a route and a name it will look through the matching parent routes to see if any have the specified name
- *
+ * Checks to see if the route requires authentication by taking a look at the route and it's parents 'meta' to see if it
+ * contains { requiresAuthentication: true }
  * @param {*} to a VueRouter Route object
- * @param {*} routeName the name of a route you're checking to see if it was matched.
- * @returns true if a matching route was found, false otherwise
+ * @returns true if the route requires authentication, false otherwise
  */
-export const routeNameMatched = (to, routeName) => {
-  return routeMatched(to, (matched) => (matched?.name === routeName));
+export const routeRequiresAuthentication = (to) => {
+  return routeMatched(to, (matched) => matched.meta?.requiresAuthentication);
```

Comment:
I decided to update this to use metadata instead of names so that we don't have to maintain a separate list of authenticated pages.

The vue3 router docs also have this exact example for authentication: https://v3.router.vuejs.org/guide/advanced/meta.html#route-meta-fields so it should migrate better.

Reply:
I like this approach, it's straightforward and the idiomatic way to handle authentication in vue router.